src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -1028,231 +991,172 @@ void TestCASESession::SessionResumptionStorage(nlTestSuite * inSuite, void * inC
         },
     };
 
-    auto & loopback = ctx.GetLoopback();
+    auto & loopback = GetLoopback();
     for (size_t i = 0; i < sizeof(testVectors) / sizeof(testVectors[0]); ++i)
     {
         auto * pairingCommissioner = chip::Platform::New<CASESession>();
         pairingCommissioner->SetGroupDataProvider(&gCommissionerGroupDataProvider);
         loopback.mSentMessageCount = 0;
-        NL_TEST_ASSERT(inSuite,
-                       gPairingServer.ListenForSessionEstablishment(&ctx.GetExchangeManager(), &ctx.GetSecureSessionManager(),
-                                                                    &gDeviceFabrics, &testVectors[i].responderStorage, nullptr,
-                                                                    &gDeviceGroupDataProvider) == CHIP_NO_ERROR);
-        ExchangeContext * contextCommissioner = ctx.NewUnauthenticatedExchangeToBob(pairingCommissioner);
+        EXPECT_EQ(gPairingServer.ListenForSessionEstablishment(&GetExchangeManager(), &GetSecureSessionManager(), &gDeviceFabrics,
+                                                               &testVectors[i].responderStorage, nullptr,
+                                                               &gDeviceGroupDataProvider),
+                  CHIP_NO_ERROR);
+        ExchangeContext * contextCommissioner = NewUnauthenticatedExchangeToBob(pairingCommissioner);
         auto establishmentReturnVal           = pairingCommissioner->EstablishSession(
-            ctx.GetSecureSessionManager(), &gCommissionerFabrics, ScopedNodeId{ Node01_01, gCommissionerFabricIndex },
+            GetSecureSessionManager(), &gCommissionerFabrics, ScopedNodeId{ Node01_01, gCommissionerFabricIndex },
             contextCommissioner, &testVectors[i].initiatorStorage, nullptr, &delegateCommissioner,
             Optional<ReliableMessageProtocolConfig>::Missing());
-        ServiceEvents(ctx);
-        NL_TEST_ASSERT(inSuite, establishmentReturnVal == CHIP_NO_ERROR);
-        NL_TEST_ASSERT(inSuite, loopback.mSentMessageCount == testVectors[i].expectedSentMessageCount);
-        NL_TEST_ASSERT(inSuite, delegateCommissioner.mNumPairingComplete == i + 1);
+        ServiceEvents();
+        EXPECT_EQ(establishmentReturnVal, CHIP_NO_ERROR);
+        EXPECT_EQ(loopback.mSentMessageCount, testVectors[i].expectedSentMessageCount);
+        EXPECT_EQ(delegateCommissioner.mNumPairingComplete, i + 1);
         SessionHolder & holder = delegateCommissioner.GetSessionHolder();
-        NL_TEST_ASSERT(inSuite, bool(holder));
-        NL_TEST_ASSERT(inSuite, holder->GetPeer() == fabricInfo->GetScopedNodeIdForNode(Node01_01));
+        EXPECT_TRUE(bool(holder));
+        EXPECT_EQ(holder->GetPeer(), fabricInfo->GetScopedNodeIdForNode(Node01_01));
         chip::Platform::Delete(pairingCommissioner);
+        gPairingServer.Shutdown();
```

Comment:
I've added this line in the test to fix the error in the next test, `Sigma1BadDestinationIdTest`. Without this modification, the SIGABRT signal was received during call `DrainAndServiceIO` in `Sigma1BadDestinationIdTest`

Reply:
@andy31415 Is it fine? Or should I fix it in another way?